httpapi/userapi/transfer_details_api_test.go

```diff
@@ -28,7 +28,7 @@ import (
 func TestHandler_CreateTransferDetail(t *testing.T) {
 	docSrv := new(testingdocuments.MockService)
 	transferSrv := new(MockTransferService)
-	srv := Service{docSrv: docSrv, transferDetailsService: transferSrv}
+	srv := Service{coreAPISrv: newCoreAPIService(docSrv), transferDetailsService: transferSrv}
```

Comment:
this test needs a mock of `coreAPISrv`, i believe we need an interface. Otherwise we are also testing the coreservice here.

Reply:
@vimukthi-git CoreAPI service will just be a wrapper for services underneath. Making this a service just adds another extra layer of interface with no usage IMO.

Reply:
usage is testing as I said above, here we can't test user service without testing the core service as well (This is only for now, there would be more uses in the future if we are creating core service for any real purpose).

Reply:
We do not need to create any interface for CoreAPIservice since its consists the interfaces and any logic would just be testable since the underneath services will mocks. I'm thinking we can cross the brodge when we really need to make it an interface. Let me know if you wanna discuss more offline :)